eth/watchers/orchestratorwatcher.go

```diff
@@ -0,0 +1,141 @@
+package watchers
+
+import (
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/eth/blockwatch"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+)
+
+const maxFutureRound = int64(1<<63 - 1)
+
+type OrchestratorWatcher struct {
+	store   orchestratorStore
+	dec     *EventDecoder
+	watcher BlockWatcher
+	lpEth   eth.LivepeerEthClient
+	quit    chan struct{}
+}
+
+func NewOrchestratorWatcher(bondingManagerAddr ethcommon.Address, watcher BlockWatcher, store orchestratorStore, lpEth eth.LivepeerEthClient) (*OrchestratorWatcher, error) {
+	dec, err := NewEventDecoder(bondingManagerAddr, contracts.BondingManagerABI)
+	if err != nil {
+		return nil, err
+	}
+
+	return &OrchestratorWatcher{
+		store:   store,
+		dec:     dec,
+		watcher: watcher,
+		lpEth:   lpEth,
+		quit:    make(chan struct{}),
+	}, nil
+}
+
+// Watch starts the event watching loop
+func (ow *OrchestratorWatcher) Watch() {
+	events := make(chan []*blockwatch.Event, 10)
+	sub := ow.watcher.Subscribe(events)
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case <-ow.quit:
+			return
+		case err := <-sub.Err():
+			glog.Error(err)
+		case events := <-events:
+			ow.handleBlockEvents(events)
+		}
+	}
+}
+
+// Stop watching for events
+func (ow *OrchestratorWatcher) Stop() {
+	close(ow.quit)
+}
+
+func (ow *OrchestratorWatcher) handleBlockEvents(events []*blockwatch.Event) {
+	for _, event := range events {
+		for _, log := range event.BlockHeader.Logs {
+			if event.Type == blockwatch.Removed {
+				log.Removed = true
+			}
+			if err := ow.handleLog(log); err != nil {
+				glog.Error(err)
+			}
+		}
+	}
+}
+
+func (ow *OrchestratorWatcher) handleLog(log types.Log) error {
+	eventName, err := ow.dec.FindEventName(log)
+	if err != nil {
+		// Noop if we cannot find the event name
+		return nil
+	}
+
+	switch eventName {
+	case "TranscoderActivated":
+		if err := ow.handleTranscoderActivated(log); err != nil {
```

Comment:
Could shorten this to just `return ow.handleTranscoderActivated(log)`

Reply:
fixed in d3fe26f